flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java

```diff
@@ -38,25 +42,57 @@
  * @param <Message> The type of the message sent between vertices along the edges.
  * @param <EdgeValue> The type of the values that are associated with the edges.
  */
-public abstract class MessagingFunction<VertexKey extends Comparable<VertexKey> & Serializable, 
-	VertexValue extends Serializable, Message, EdgeValue extends Serializable> implements Serializable {
+public abstract class MessagingFunction<VertexKey extends Comparable<VertexKey> & Serializable,
+		VertexValue extends Serializable, Message, EdgeValue extends Serializable> implements Serializable {
 
 	private static final long serialVersionUID = 1L;
-	
+
+	// --------------------------------------------------------------------------------------------
+	//  Attributes that allow vertices to access their in/out degrees and the total number of vertices
+	//  inside an iteration.
+	// --------------------------------------------------------------------------------------------
+
+	private long numberOfVertices = -1L;
+
+	public long getNumberOfVertices() throws Exception{
+		if (numberOfVertices == -1) {
+			throw new InaccessibleMethodException("The number of vertices option is not set");
```

Comment:
Can we make the error message a bit more informative? i.e. explain how to set the option?